lib/srv/app/transport.go

```diff
@@ -170,9 +173,28 @@ func (t *transport) rewriteRequest(r *http.Request) error {
 	r.Header.Set(teleport.AppJWTHeader, t.c.jwt)
 	r.Header.Set(teleport.AppCFHeader, t.c.jwt)
 
+	// Add headers from rewrite configuration.
+	if t.c.rewrite != nil && len(t.c.rewrite.Headers) > 0 {
+		t.rewriteHeaders(r)
+	}
+
 	return nil
 }
 
+// rewriteHeaders applies headers rewrites from the application configuration.
+func (t *transport) rewriteHeaders(r *http.Request) {
+	for _, header := range t.c.rewrite.Headers {
+		values, err := services.ApplyValueTraits(header.Value, t.c.traits)
+		if err != nil {
+			logrus.Warnf("Failed to apply traits to %q: %v.", header.Value, err)
+			continue
```

Comment:
why not treat this as an error?

Reply:
I don't think this should be an error - the most common scenario this would fail is if there's no such trait which I imagine wouldn't be very rare so we probably don't want to make this a hard failure. We treat missing trait same way elsewhere as well, just log it.